From 2029f288d3f615d750c3c1ff65f28797a2220dd5 Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Mon, 15 Jan 2018 11:47:20 -0500
Subject: [PATCH 19/19] Move HVM self permissions into interface

Since all HVMs will need these permissions, place them in the interface
defining them.

Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
---
 policy/modules/xen/guesthvm.te | 5 -----
 policy/modules/xen/xen.if      | 5 +++++
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/policy/modules/xen/guesthvm.te b/policy/modules/xen/guesthvm.te
index f8f5cf1..fbf117d 100644
--- a/policy/modules/xen/guesthvm.te
+++ b/policy/modules/xen/guesthvm.te
@@ -71,11 +71,6 @@ nilfvm_event_src(hvm_guest_t, nilfvm-hvm_guest_evchn_t)
 # HVM Guest Local policy
 #
 
-allow hvm_guest_t self:mmu physmap;
-allow hvm_guest_t self:grant copy;
-allow hvm_guest_t self:resource remove;
-allow hvm_guest_t self:domain2 get_vnumainfo;
-
 ndvm_use(hvm_guest_t)
 nilfvm_use(hvm_guest_t)
 dom0_copy_grant(hvm_guest_t)
diff --git a/policy/modules/xen/xen.if b/policy/modules/xen/xen.if
index a5be9a3..3ab1c56 100644
--- a/policy/modules/xen/xen.if
+++ b/policy/modules/xen/xen.if
@@ -649,6 +649,11 @@ interface(`xen_hvm_type',`
                 attribute hvm_type;
 	')
 
+	allow $1 self:mmu physmap;
+	allow $1 self:grant copy;
+	allow $1 self:resource remove;
+	allow $1 self:domain2 get_vnumainfo;
+
 	typeattribute $1 hvm_type;
 ')
 ########################################
-- 
2.1.4

